src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java

```diff
@@ -244,7 +253,8 @@ private boolean executeCmd(List<String> cmds) throws IOException, InterruptedExc
         }
 
         private Result executeCmd(String cmd) throws IOException, InterruptedException {
-            PrintStream stdout = listener.getLogger();
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            TeeOutputStream stdout = new TeeOutputStream(listener.getLogger(), baos);
```

Comment:
stdout can't be closed, it would close listener.getLogger() too, and it does nothing special other than closing each of the two streams